### PR TITLE
refactor the mixer threads

### PR DESCRIPTION
**Pull request by shuchitak**

### CODE REVIEW

File: lib_xua/doc/rst/images/mixer.png
Comment (xross) [inline lib_xua/doc/rst/images/mixer.png:1]: Is it possible to add the source for these diagrams?

File: lib_xua/lib_build_info.cmake [hunk 1]
```diff
@@ -27,7 +27,7 @@ set(LIB_DEPENDENT_MODULES "lib_adat(1.2.0)"
                           "lib_spdif(6.1.0)"
                           "lib_sw_pll(2.2.0)"
                           "lib_xassert(4.2.0)"
-                          "lib_xud(2.3.1)")
+                          "lib_xud(develop)")
```
Comment (xross) [inline lib_xua/lib_build_info.cmake:30]: Do we want to do this? or leave for now, release lib_xud then bump?

File: lib_xua/src/core/mixer/fastmix.S [hunk 2]
```diff
@@ -36,19 +36,16 @@ doMix##i##: ;\
...
-          mov       r0, r1;\
-          ldc       r2, 0x19;\
-          sext      r0, r2;\
-          eq        r0, r0, r1;\
-          bf        r0, .L20; \
-\
-          shl       r0, r1, 0x7;\
+          ldc       r2, 0x19; \
+          lsats     r1, r0, r2; \
+          shl       r0, r1, 7; \
```
Comment (xross) [inline lib_xua/src/core/mixer/fastmix.S:48]: much nicer :)

File: lib_xua/lib_build_info.cmake (hunk 1 above)
Comment (shuchitak) [reply lib_xua/lib_build_info.cmake:30]: The latest lib_xud has the RX rise delay change https://github.com/xmos/lib_xud/commit/039cd2b8d20f912cd17aa5ccb3e77c7b90227388 which is required for MIDI tests to pass on sw_usb_audio so I think we take the head of develop of lib_xud

Comment (xross) [reply lib_xua/lib_build_info.cmake:30]: I've just made 2.3.2 release of lib_xud :)

File: lib_xua/lib_build_info.cmake [hunk 3]
```diff
@@ -27,7 +27,7 @@ set(LIB_DEPENDENT_MODULES "lib_adat(1.2.0)"
                           "lib_spdif(6.1.0)"
                           "lib_sw_pll(2.2.0)"
                           "lib_xassert(4.2.0)"
-                          "lib_xud(2.3.1)")
+                          "lib_xud(2.3.2)")
```
Comment (xross) [inline lib_xua/lib_build_info.cmake:30]: I've does the pin to the new 2.3.2 release

Comment (shuchitak) [reply lib_xua/lib_build_info.cmake:30]: thanks!

**Review by xross — DISMISSED** [review]

**Review by xross — APPROVED** [review]

**Pull request merged**